File: src/main/kotlin/GitHubRepoApiImpl.kt

import io.ktor.client.*
import io.ktor.client.call.*
import io.ktor.client.request.*

class GitHubRepoApiImpl(
    private val httpClient: HttpClient
): GitHubRepoApi {
    override suspend fun getRepoFiles(repoOwner: String, repoName: String): List<GitHubContent> {
        return httpClient.get("https://api.github.com/repos/${repoOwner}/${repoName}/contents/").body()
    }
}

File: dart/lib/github_repo_api_impl.dart
import 'dart:convert' show jsonDecode;

import 'package:http/http.dart' as http;

import 'github_content/github_content.dart';
import 'github_repo_api.dart';

class GitHubRepoApiImpl implements GitHubRepoApi {
  @override
  Future<Iterable<GitHubContent>> getRepoFiles({required String repoOwner, required String repoName}) async {
    final response = await http.get(Uri.parse('https://api.github.com/repos/$repoOwner/$repoName/contents/'));
    return (jsonDecode(response.body) as List<dynamic>).map((e) => GitHubContent.fromJson(e));
  }
  
}